stdlib/sql.go

```diff
@@ -0,0 +1,171 @@
+// Package stdlib provides a compatability layer from gorqlite to database/sql.
+package stdlib
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+	"io"
+
+	"github.com/rqlite/gorqlite"
+)
+
+func init() {
+	sql.Register("rqlite", &Driver{})
+}
+
+type Driver struct{}
+
+func (d *Driver) Open(name string) (driver.Conn, error) {
+	conn, err := gorqlite.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	return &Conn{conn}, nil
+}
+
+type Conn struct {
+	*gorqlite.Connection
+}
+
+func (c *Conn) Prepare(query string) (driver.Stmt, error) {
+	return &Stmt{Stmt: query, Conn: c}, nil
+}
+
+func (c *Conn) Close() error {
+	c.Connection.Close()
+	return nil
+}
+
+func (c *Conn) Begin() (driver.Tx, error) {
```

Comment:
Should the return something like "not supported" since rqlite doesn't support traditional transactions?

Reply:
Many operations in database/sql call this function, so it must return a usable value and not an error. It does not actually do anything, since rqlite does not support this kind of transaction, but it must be there as a placeholder that satisfies interfaces. In other words, as soon as Exec or Query is called, changes are instantly made, so all of the transaction functions are just no-ops, but they must exist.

This is the principle behind this PR: database/sql is an unideal wrapper for rqlite (which leads to many no-op functions), but it is still a perfectly functional wrapper. Therefore, given that it is the widely accepted standard in the Go community that everything else integrates with, I think that it makes sense to provide a compatibility layer for it.